syntax = "proto3";

import "google/protobuf/empty.proto";

package samples.hello.guidance.messages;

// Ground mode configuration
message Config {
    bool say = 1;
    bool my_guidance_config = 2;
}

// Union of all possible commands of this package.
message Command {
    oneof id {
        google.protobuf.Empty empty = 1;
        google.protobuf.Empty my_guidance_command = 2;
    }
}

// Union of all possible events of this package.
message Event {
    oneof id {
        uint32 count = 1;
        uint32 my_guidance_event = 2;
        google.protobuf.Empty mission_complete = 3;
        google.protobuf.Empty path_obstructed = 4;
        google.protobuf.Empty goto_complete = 5;
    }
}

message FindTowerParameters_Guidance {
    float x = 1;
    float y = 2;
    bool visible = 3;
}


message GoTo_Guidance {
    float x = 1;
    float y = 2;
    float z = 3;
    float yaw = 4;
    float speed = 5;
}